qbraid/transpiler/conversions/openqasm3/openqasm3_to_cudaq.py

```diff
@@ -0,0 +1,188 @@
+
+    module.unroll()
+    program = module.unrolled_ast
+
+    kernel: PyKernel = cudaq.make_kernel()
+    ctx: dict[str, Optional[QuakeValue]] = {}
+    gate_kernels: dict[str, PyKernel] = {}
+
+    def get_gate(name: str, targs: tuple[type]) -> PyKernel:
+        if name in gate_kernels:
+            return gate_kernels[name]
+
+        gate_kernels[name] = make_gate_kernel(name, targs)
+        return gate_kernels[name]
+
+    def qubit_lookup(qubit: ast.IndexedIdentifier | ast.Identifier) -> QuakeValue:
+        assert isinstance(
+            qubit, ast.IndexedIdentifier
+        ), f"all identifiers should've been indexed: {qubit}"
+
+        assert len(qubit.indices) == 1, f"multi-dim arrays are not supported: {qubit.indices}"
+
+        inds = qubit.indices[0]
+
+        assert len(inds) == 1, f"indices should've been a single integer: {inds}"
+        assert isinstance(ind := inds[0], ast.IntegerLiteral)
+
+        q = ctx[qubit.name.name][ind.value]
+        return q
+
+    # pylint: disable-next=too-many-nested-blocks
+    for statement in program.statements:
+        if isinstance(statement, ast.Include):
+            if statement.filename not in {"stdgates.inc", "qelib1.inc"}:
+                raise ProgramConversionError(f"Custom includes are unsupported: {statement}")
+        elif isinstance(statement, ast.QubitDeclaration):
+            ctx[statement.qubit.name] = kernel.qalloc(statement.size.value)
+        elif isinstance(statement, ast.ClassicalDeclaration):
+            if statement.init_expression and isinstance(
+                statement.init_expression, ast.QuantumMeasurement
+            ):
+                ctx[statement.identifier.name] = kernel.mz(
+                    qubit_lookup(statement.init_expression.qubit)
+                )
+            else:
+                ctx[statement.identifier.name] = None
+        elif isinstance(statement, ast.QuantumMeasurementStatement):
+            val = kernel.mz(qubit_lookup(statement.measure.qubit))
+            if statement.target is not None:
+                assert isinstance(
+                    statement.target, ast.IndexedIdentifier
+                ), f"identifiers should've been unrolled to indexed identifiers: {statement.target}"
+                ctx[statement.target.name.name] = val
+        elif isinstance(statement, ast.QuantumGate):
+            name, qubits = statement.name.name, statement.qubits
+
+            args = []
+            for arg in statement.arguments:
+                assert arg.value is not None, f"gate arguments should've been literals: {arg}"
+                args.append(arg.value)
+            targs = [type(a) for a in args]
+
+            qubit_refs = [qubit_lookup(q) for q in qubits]
+
+            # pyqasm unrolls multiple modifiers.
+            # ctrl isn't supported so multi-ctrl is not an issue at the moment.
+            assert len(statement.modifiers) < 1
+
+            if len(statement.modifiers) == 1:
```

Comment:
Would this case occur if we're asserting the length to be < 1? Or maybe the assertion is supposed to be for `len(statement.modifiers) <= 1`?